### PR TITLE
DR | Full name missing alert update

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > Show missing full name alert work resulted in:
  > - Moving missing info & needs to be verified alerts to shared
  > - Create a shared component to render the appropriate alert or the save-in-progress intro component
  > - Change placement of in progress content: unauth full alert only renders at the bottom, authenticated in progress alert renders only at top
  > - Update tests

- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#81440](https://github.com/department-of-veterans-affairs/va.gov-team/issues/81440)

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

| New changes | NOD | HLR | SC |
| ------- | ------ | ----- | ----- |
| Original unauth | unauth_old_decision-reviews_board-appeal_request-board-appeal-form-10182_introduction | unauth_old_decision-reviews_higher-level-review_request-higher-level-review-form-20-0996_introduction | unauth_old_decision-reviews_supplemental-claim_file-supplemental-claim-form-20-0995_introduction |
| New unauth |  unauth_decision-reviews_board-appeal_request-board-appeal-form-10182_introduction | unauth_decision-reviews_higher-level-review_request-higher-level-review-form-20-0996_introduction | unauth_decision-reviews_supplemental-claim_file-supplemental-claim-form-20-0995_introduction |

| No visual change | NOD | HLR | SC |
| ------- | ------ | ----- | ----- |
| In progress | sip_decision-reviews_board-appeal_request-board-appeal-form-10182_introduction | sip_decision-reviews_higher-level-review_request-higher-level-review-form-20-0996_introduction | sip_decision-reviews_supplemental-claim_file-supplemental-claim-form-20-0995_introduction |
| New form | new_decision-reviews_board-appeal_request-board-appeal-form-10182_introduction | new_decision-reviews_higher-level-review_request-higher-level-review-form-20-0996_introduction | new_decision-reviews_supplemental-claim_file-supplemental-claim-form-20-0995_introduction |
| Missing info | missing_decision-reviews_board-appeal_request-board-appeal-form-10182_introduction | missing_decision-reviews_higher-level-review_request-higher-level-review-form-20-0996_introduction | missing_decision-reviews_supplemental-claim_file-supplemental-claim-form-20-0995_introduction |
| Needs verified | loa1_decision-reviews_board-appeal_request-board-appeal-form-10182_introduction | loa1_decision-reviews_higher-level-review_request-higher-level-review-form-20-0996_introduction | loa1_decision-reviews_supplemental-claim_file-supplemental-claim-form-20-0995_introduction |

## What areas of the site does it impact?

All Decision Review forms: HLR, NOD & SC

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/shared/components/ShowAlertOrSip.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    ].filter(Boolean);
+    const missing = readableList(missingList);
+    return bottom ? (
+      <div className={classes} />
+    ) : (
+      <NeedsMissingInfoAlert missing={missing} />
+    );
+  }
+
+  return !loggedIn && !bottom ? null : (
```
Comment (anniebtran) [inline src/applications/appeals/shared/components/ShowAlertOrSip.jsx:59]: Small nit — wondering if separating the ternary into separate returns could make this a little more readable. i.e.
```
if (!loggedIn && !bottom) return null;

return (<div .... )
```
If we did that, then I also wonder if you could return `null` earlier in the function if someone isn't logged in and `bottom` is false/nil? 🤔

**Review by anniebtran — COMMENTED** [review]
Looking great! Just a couple of questions

File: src/applications/appeals/10182/containers/IntroductionPage.jsx [hunk 2]
```diff
@@ -15,142 +12,133 @@ import {
...
-        // don't need to pass the entire formConfig
-        customText,
-      },
-      ariaDescribedby: 'main-content',
-    };
+  const { route, location } = props;
+  const { formConfig, pageList } = route;
+  const { formId, prefillEnabled, savedFormMessages, downtime } = formConfig;
+  const sipOptions = {
+    // ariaDescribedby: 'main-content',
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/10182/containers/IntroductionPage.jsx:32]: ;tldr: This is a placeholder to remind us that the current save-in-progress buttons don't have any descriptive text

There is a [Slack discussion](https://dsva.slack.com/archives/C04KW0B46N5/p1716415144602809) about this from last week. Basically, the sign in and continue this form button inside the save-in-progress intro component need to point to related content so a screenreader users knows there is important context that goes along with the button. Previously, when we used a plain `button`, the above option would add an `aria-describedby` to the button pointing to the intro page header above the process list. Now that we've updated all plain `button`s to use `va-button`, the only option we're left with is to use the [`message-aria-describedby` property](https://design.va.gov/storybook/?path=/docs/uswds-va-button--primary) which only accepts a plain string and not an ID. We'll go with whatever recommendation is provided in that Slack thread

File: src/applications/appeals/shared/components/ShowAlertOrSip.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/shared/components/ShowAlertOrSip.jsx]: Good idea! Updating the code now.

File: src/applications/appeals/10182/containers/IntroductionPage.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (anniebtran) [reply src/applications/appeals/10182/containers/IntroductionPage.jsx:34]: Ohhhhhh okay. Thank you for the clarification 🙏

**Review by anniebtran — APPROVED** [review]
LGTM!

**Pull request merged**